numpy/core/src/multiarray/mapping.c

```diff
@@ -3374,6 +3374,7 @@ PyArray_MapIterArray(PyArrayObject * a, PyObject * index)
 static void
 arraymapiter_dealloc(PyArrayMapIterObject *mit)
 {
+    PyArray_ResolveWritebackIfCopy(mit->array);
```

Comment:
I am slightly uneasy about putting this here, but I guess it's OK.

The problem is that we are once again tying writeback to a dealloc, which is what caused u problems in PyPy before. But I think it's probably OK here because mapiter objects are only created and managed in C code, internally in numpy, so I think they are always deallocated immediately when no longer needed.

Reply:
However, I remember reading some comments here that one or two downstream projects use mapiters, even though they're not supposed to. May need to warn about this in the release note. I'll address it in a comment later about documentation.